ambry-api/src/main/java/com/github/ambry/named/NamedBlobDb.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ */
+
+package com.github.ambry.named;
+
+import com.github.ambry.account.Container;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+
+public interface NamedBlobDb {
+  CompletableFuture<NamedBlobRecord> get(String accountName, String containerName, String blobName);
```

Comment:
Are the methods going to throw only unchecked exceptions on errors?

Reply:
Checked exceptions will be communicated asynchronously using the `CompletableFuture` to simplify the error handling path. So, the synchronous part of this will only throw unchecked exceptions (on coding bugs), or complete the future with a checked exception (on system/user errors).